File: java/backend-spring/src/main/java/com/countriesdb/examples/spring/ValidationController01.java
package com.countriesdb.examples.spring;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;

@RestController
@RequestMapping("/api/validate")
public class ValidationController01 {
    private final CountriesDbClient client;

    public ValidationController01(CountriesDbClient client) {
        this.client = client;
    }

    // 1. Basic country validation
    @PostMapping("/country/basic")
    public ResponseEntity<?> validateBasicCountry(@RequestBody Map<String, String> payload) {
        String country = payload.get("country");
        if (country == null || country.isBlank()) {
            return ResponseEntity.badRequest().body(Map.of(
                "valid", false,
                "message", "Country code is required"
            ));
        }
        
        Map<String, Object> result = client.validateCountry(Map.of("code", country));
        
        if (!Boolean.TRUE.equals(result.get("valid"))) {
            return ResponseEntity.unprocessableEntity().body(result);
        }
        
        return ResponseEntity.ok(Map.of(
            "valid", true,
            "message", "Country validated",
            "payload", payload
        ));
    }
}

File: csharp/backend-aspnetcore/Controllers/ValidationController01.cs
using Microsoft.AspNetCore.Mvc;

namespace BackendAspNetCore.Controllers;

[ApiController]
[Route("api/validate")]
public class ValidationController01 : ControllerBase
{
    private readonly CountriesDbClient _client;

    public ValidationController01(CountriesDbClient client)
    {
        _client = client;
    }

    // 1. Basic country validation
    [HttpPost("country/basic")]
    public async Task<IActionResult> ValidateBasicCountry([FromBody] Dictionary<string, string> payload)
    {
        var country = payload.GetValueOrDefault("country");
        if (string.IsNullOrWhiteSpace(country))
        {
            return BadRequest(new Dictionary<string, object>
            {
                ["valid"] = false,
                ["message"] = "Country code is required"
            });
        }

        var result = await _client.ValidateCountryAsync(new Dictionary<string, object>
        {
            ["code"] = country
        });

        if (!result.Valid)
        {
            return UnprocessableEntity(result);
        }

        return Ok(new Dictionary<string, object>
        {
            ["valid"] = true,
            ["message"] = "Country validated",
            ["payload"] = payload
        });
    }
}

